Sources/Brave/Frontend/UserContent/UserScripts/__firefox__.js

```diff
@@ -369,15 +369,29 @@ if (!window.__firefox__) {
     return result;
   };
   
+  $.dispatchEvent = function(event) {
+    let originalDispatchEvent = window.dispatchEvent(event);
+    return originalDispatchEvent;
```

Comment:
@StephenHeaps this doesn't solve the issue I'm afraid. When `$.dispatchEvent` function is called it assigns `window.dispatchEvent(event)` to `originalDispatchEvent`, but at this point `window.dispatchEvent` can be already modified. 
We should assign 
`originalDispatchEvent = window.dispatchEvent` somewhere at the document start. Same for `addEventListener `

Reply:
Thanks, similar to `window.webkit`, `window.dispatchEvent` and `window.addEventListener` are re-initialized by WebKit when deleted:

iOS 16 | iOS 15
--|--
<img width="796" alt="iOS 16" src="https://github.com/brave/brave-ios/assets/5314553/e59e9152-d652-4e7b-a8fa-b0f3cfac05dd"> | <img width="796" alt="iOS 15" src="https://github.com/brave/brave-ios/assets/5314553/1063f408-cd8c-4a02-97bc-7803ec45a67b">

I've updated `$.dispatchEvent` and `$.addEventListener` to delete these functions prior to use similar to `$.postNativeMessage` implementation above for consistency. If declaring `originalDispatchEvent` / `originalAddEventListener` earlier in `__firefox__.js` is preferred we should be able to go that route also :).